src/fx/AnimateableTimeStretchPitchScaleReader.cpp

```diff
@@ -0,0 +1,43 @@
+
+
+/*******************************************************************************
+ * Copyright 2009-2025 Jörg Müller
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+#include "fx/AnimateableTimeStretchPitchScaleReader.h"
+
+#include "IReader.h"
+
+AUD_NAMESPACE_BEGIN
+
+AnimateableTimeStretchPitchScaleReader::AnimateableTimeStretchPitchScaleReader(std::shared_ptr<IReader> reader, std::shared_ptr<AnimateableProperty> timeStretch,
+                                                                               std::shared_ptr<AnimateableProperty> pitchScale, StretcherQuality quality,
+                                                                               bool preserveFormant) :
+    TimeStretchPitchScaleReader(reader, 1.0, 1.0, quality, preserveFormant), m_timeStretch(timeStretch), m_pitchScale(pitchScale)
```

Comment:
In order to properly initialize the stretcher, I would read the value at the beginning of the property instead of setting them to 1 here, otherwise if it doesn't start at 1, you immediately have a jump. The same could be useful to do in the seek function, since it resets and reinitializes the stretcher.